### PR TITLE
update evmos to v20

### PR DESCRIPTION
**Pull request by mn13**
## Summary by CodeRabbit

- **New Features**
	- Added support for new SDK fields in CLI commands.
	- Introduced rate limiting and ERC20 token management functionalities.
	- Enhanced genesis commands with validation for bech32 formats.
	- New filter parameter for the `Rules` query.
	- Added a `Nonce` field to the Space to prevent race conditions.
	- Updated keychain metadata with new fields: `Url` and `Keybase Id`.

- **Bug Fixes**
	- Resolved issues with default parameters in genesis.
	- Fixed keychain fee handling and metadata updates.

- **Chores**
	- Updated dependencies to the latest versions, including Evmos SDK and Cosmos SDK.
	- Enhanced local blockchain setup script for better functionality.

### CODE REVIEW

File: warden/app/legacy.go [hunk 1]
```diff
@@ -366,6 +393,7 @@ func RegisterLegacyModules(registry cdctypes.InterfaceRegistry) map[string]appmo
 		gmptypes.ModuleName:         gmpmodule.AppModule{},
 		wasmtypes.ModuleName:        wasm.AppModule{},
 		evmtypes.ModuleName:         evm.AppModule{},
+		erc20types.ModuleName:       erc20.AppModule{},
```
Comment (backsapc) [inline warden/app/legacy.go]: Do we need this?

File: warden/app/legacy.go [hunk 2]
```diff
@@ -329,6 +339,23 @@ func (app *App) registerLegacyModules(appOpts servertypes.AppOptions, wasmOpts [
...
+	// Create the rate limit keeper
+	app.RateLimitKeeper = *ratelimitkeeper.NewKeeper(
+		app.appCodec,
+		runtime.NewKVStoreService(app.GetKey(ratelimittypes.StoreKey)),
+		app.GetSubspace(ratelimittypes.ModuleName),
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		app.BankKeeper,
+		app.IBCKeeper.ChannelKeeper,
+		app.IBCKeeper.ChannelKeeper, // ICS4Wrapper
+	)
```
Comment (backsapc) [inline warden/app/legacy.go]: Looks like we don't need this

File: warden/app/app.go [hunk 3]
```diff
@@ -164,6 +168,8 @@ type App struct {
 	// evmOS
 	EvmKeeper       *evmkeeper.Keeper
 	FeeMarketKeeper feemarketkeeper.Keeper
+	Erc20Keeper     erc20keeper.Keeper
+	RateLimitKeeper ratelimitkeeper.Keeper
```
Comment (backsapc) [inline warden/app/app.go]: Looks like we don't need this

File: warden/app/legacy.go (hunk 1 above)
Comment (mn13) [reply warden/app/legacy.go]: Hm, you're right, we can ommit erc20 module now and add it later when will use it's precompiles

**Review by backsapc — APPROVED** [review]

**Pull request merged**